src/mappings/referrer.ts

```diff
@@ -17,54 +19,69 @@ export function handleReferralCodeCreated(event: OnReferralCodeCreated): void {
 }
 
 export function handleReferralCodeUpserted(event: OnReferralCodeUpserted): void {
-    let existingReferralCode = getReferralCode(event.params.oldReferralCode)
-    let trader = getOrCreateTrader(event.params.addr)
-    // Only proceed any action, if the referee has a code attached
-    if (existingReferralCode) {
-        // Referee removes code
-        if (event.params.action == 1) {
-            // trader no longer has a referee code
-            trader.refereeCode = null
-            trader.save()
-
-            // update referee count for code
-            existingReferralCode.numReferees = existingReferralCode.numReferees.minus(BI_ONE)
-            existingReferralCode.save()
-            return
-        }
+    if (!event.params.oldReferralCode) {
```

Comment:
I found the contract source code here: https://github.com/perpetual-protocol/perp-referral-contracts/blob/master/contracts/PerpetualProtocolReferrer.sol

`event.params.oldReferralCode` should always be not null even if it is ADD / REPLACE.

Reply:
Thanks! The source code is really helpful. The empty string check has been removed.